src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/MergePropertiesResourceTransformer.kt

```diff
@@ -0,0 +1,137 @@
+package com.github.jengelman.gradle.plugins.shadow.transformers
+
+import com.github.jengelman.gradle.plugins.shadow.internal.WithPatternFilterable
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.util.LinkedHashSet
+import java.util.Properties
+import javax.inject.Inject
+import org.apache.tools.zip.ZipEntry
+import org.apache.tools.zip.ZipOutputStream
+import org.gradle.api.GradleException
+import org.gradle.api.file.FileTreeElement
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.util.PatternFilterable
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * Transformer to merge the contents of equally named property files. The set of property files to
+ * merge is determined by the implemented [PatternFilterable].
+ *
+ * Conflicting property values from different property files are detected and lead to an error. Use
+ * the [ignoreDuplicates] property to specify path patterns for property files that should skip the
+ * conflicting values check.
+ */
+@CacheableTransformer
+public open class MergePropertiesResourceTransformer @Inject constructor(objectFactory: ObjectFactory) :
```

Comment:
Seems duplicate of https://github.com/GradleUp/shadow/blob/9ada0afe7bafd00df68f286978be09daf82a750f/src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/PropertiesFileTransformer.kt#L101.

Reply:
:facepalm: